lib/wanda/policy.ex

```diff
@@ -3,43 +3,44 @@ defmodule Wanda.Policy do
   Handles integration events.
   """
 
-  alias Cloudevents.Format.V_1_0.Event, as: CloudEvent
-  alias Wanda.Execution.{Fact, Target}
+  alias Trento.Checks.V1.{
+    ExecutionRequested,
+    FactsGathered
+  }
 
-  @execution_requested_event "trento.checks.v1.ExecutionRequested"
-  @facts_gathered_event "trento.checks.v1.FactsGathered"
+  alias Wanda.Execution.{
+    Fact,
+    Target
+  }
 
-  # TODO: move the CloudEvent unwrapping part in the contract repository.
-  # we want to receive just the domain event here
-
-  @spec handle_event(CloudEvent.t()) :: :ok | {:error, any}
-  def handle_event(%CloudEvent{
-        type: @execution_requested_event,
-        data: %{"execution_id" => execution_id, "group_id" => agent_id, "targets" => targets}
+  @spec handle_event(ExecutionRequested.t() | FactsGathered.t()) :: :ok | {:error, any}
+  def handle_event(%ExecutionRequested{
+        execution_id: execution_id,
+        group_id: agent_id,
+        targets: targets
       }) do
     execution_impl().start_execution(
       execution_id,
       agent_id,
-      Enum.map(targets, fn %{"agent_id" => agent_id, "checks" => checks} ->
+      Enum.map(targets, fn %{agent_id: agent_id, checks: checks} ->
         %Target{agent_id: agent_id, checks: checks}
       end)
     )
   end
 
-  def handle_event(%CloudEvent{
-        type: @facts_gathered_event,
-        data: %{"execution_id" => execution_id, "agent_id" => agent_id, "facts" => facts}
+  def handle_event(%FactsGathered{
+        execution_id: execution_id,
+        agent_id: agent_id,
+        facts_gathered: facts_gathered
       }) do
     execution_impl().receive_facts(
       execution_id,
       agent_id,
-      Enum.map(facts, fn %{"check_id" => check_id, "name" => name, "value" => value} ->
+      Enum.map(facts_gathered, fn %{check_id: check_id, name: name, value: value} ->
```

Comment:
It looks like the usage of `oneof` in elixir is much simpler than in golang, as I see that you can simply use the `value` field, without casting the the different options. It looks super nice.

Reply:
when building the event you need to pass a tuple `{:type, struct}`, but when getting the value back you get the pure struct it seems